### PR TITLE
Ethans branch

### PR DESCRIPTION
**Pull request by ethantanen**

### CODE REVIEW

File: js/graphData.js [hunk 1]
```diff
@@ -1,181 +1,149 @@
...
-            unix_time: unix_time,
-            date: date,
-            soil_temp: soil_temp,
-            soil_moisture: soil_moisture,
-            air_temp: air_temp,
-            air_humidity: air_humidity,
-            heat_index: heat_index,
-            sunlight: sunlight,
-            watered_today: watered_today
-        };
```
Comment (ethantanen) [inline js/graphData.js:138]: If the key has the same name as the value JavaScript will automatically make the key the variables name so:`{key_name: key_name}` can become: `{key_name}`

File: js/graphData.js [hunk 2]
```diff
@@ -1,181 +1,149 @@
...
-    soil_moisture.push("Soil Moisture (%)");
-    air_temp.push("Air Temp (°C)");
-    air_humidity.push("Humidity (%)");
-    heat_index.push("Heat Index (°C)");
-    sunlight.push("Sunlight (%)");
-    watered_today.push("Watered Today (0/1)");
-    
-    
-    try {
-        $.get("../databaseconnect.php",
```
Comment (ethantanen) [inline js/graphData.js:92]: You were pretty close to getting this to work. So $.get() is asynchronous meaning the programming continues to run while this function makes a network request. So the return statement below was executing before the data was retrieved from your php server.

File: js/graphData.js [hunk 3]
```diff
@@ -1,181 +1,149 @@
...
-                soil_moisture.push(data[i].soil_moisture);
-                air_temp.push(data[i].air_temp);
-                air_humidity.push(data[i].air_humidity);
-                heat_index.push(data[i].heat_index);
-                sunlight.push(data[i].sunlight);
-                watered_today.push(data[i].watered_today);
-            }
-            // alert("Second success");
-        })
-        .fail(function() {
```
Comment (ethantanen) [inline js/graphData.js:124]: The error thrown for the fail block here will be handled by the catch block in the updated code

File: js/graphData.js [hunk 4]
```diff
@@ -1,181 +1,149 @@
...
-      ["Soil Temp (°C)", 0],
-      ["Soil Moisture (%)", 0],
-      ["Air Temp (°C)", 0],
-      ["Humidity (%)", 0],
+    columns: [
+      ['x', 0], //,'2020/8/18 1:35:45','2020/8/18 1:41:45']
+      ['Soil Temp (°C)', 0],
+      ['Soil Moisture (%)', 0],
+      ['Air Temp (°C)', 0],
+      ['Humidity (%)', 0],
```
Comment (ethantanen) [inline js/graphData.js:16]: Consider making these strings constants if you're going to reuse them below.

**Comment by ethantanen** [comment]
My IDE automatically added the spacing, double quotations and commas just fyi...the way you were doing it was fine though.

File: js/graphData.js [hunk 5]
```diff
@@ -1,181 +1,149 @@
...
+  soil_moisture.push('Soil Moisture (%)');
+  air_temp.push('Air Temp (°C)');
+  air_humidity.push('Humidity (%)');
+  heat_index.push('Heat Index (°C)');
+  sunlight.push('Sunlight (%)');
+  watered_today.push('Watered Today (0/1)');
+
+  try {
+    // const data = await $.get('../databaseconnect.php');
+    const data = await $.get('http://localhost:3005'); // <-- use this if you're using the random data from my node script
```
Comment (ethantanen) [inline js/graphData.js:87]: dont forget to switch this back to the php server when the time comes!!

File: node/app.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+const MAX_RANDOM_NUMBER = 100;
+const NUMBER_OF_DATA_PTS = 15;
+
+const getRandomData = () => {
+  const getRandomNumber = () => Math.random() * MAX_RANDOM_NUMBER;
+
+  const data = [];
+  for (let i = 0; i < NUMBER_OF_DATA_PTS; i += 1) {
+    data.push({
+      date: i,
```
Comment (ethantanen) [inline node/app.js:19]: the date is the only value that needs to monotonically ascend

**Review by lukestroh — APPROVED** [review]

**Pull request merged**